File: little-tools/json/src/main/java/com/quane/little/tools/json/JSONSerializer.java
package com.quane.little.tools.json;

import java.io.IOException;
import java.io.StringWriter;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.module.scala.DefaultScalaModule;

public class JSONSerializer {

	private final ObjectMapper mapper;

	public JSONSerializer() {
		mapper = new ObjectMapper();
		mapper.registerModule(new DefaultScalaModule());
		mapper.enableDefaultTyping();
	}

	public String serialize(Object value) throws IOException {
		StringWriter writer = new StringWriter();
		mapper.writeValue(writer, value);
		return writer.toString();
	}

}

File: little-tools/json/src/main/scala/com/quane/little/tools/json/JSONSerializer.scala
package com.quane.little.tools.json

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.scala.DefaultScalaModule

class JSONSerializer {

  val mapper = new ObjectMapper
  mapper.registerModule(new DefaultScalaModule)
  mapper.setDefaultTyping(new LanguageTypeResolverBuilder)

  def serialize(value: Any): String = mapper.writeValueAsString(value)

}